src/main/java/it/pagopa/pm/gateway/controller/CreditCardPaymentController.java

```diff
@@ -105,10 +97,12 @@ public ResponseEntity<String> resumeCreditCardPaymentStep2(@RequestHeader(requir
                                                                @PathVariable String requestId) {
         log.info("START - POST {}{} info for requestId: {}", REQUEST_PAYMENTS_VPOS, REQUEST_PAYMENTS_RESUME_CHALLENGE, requestId);
         MdcUtils.setMdcFields(mdcFields);
-        String urlRedirect = StringUtils.join(responseUrlRedirect, requestId);
-        resumeStep2Service.startResumeStep2(requestId);
+
+        String clientReturnUrl = resumeStep2Service.startResumeStep2(requestId);
```

Comment:
why is `clientReturnUrl` a return value of `startResumeStep2`? it is a configuration that can be retrieved directly from the `clientConfig`

Reply:
Because I need the clientId to extract the correct configuration and the query to retrieve the clientId is executed inside the service.